lib/matplotlib/scale.py

```diff
@@ -458,6 +458,117 @@ def get_transform(self):
+
+    For values close to zero, this is essentially a linear scale,
+    but for larger values (either positive or negative) it is asymptotically
+    logarithmic. The transition between these linear and logarithmic regimes
+    is smooth, and has no discontinuities in the function gradient
+    in contrast to the `.SymmetricalLogScale` ("symlog") scale.
+
+    Specifically, the transformation of an axis coordinate :math:`a` is
+    :math:`a \\rightarrow a_0 \\sinh^{-1} (a / a_0)` where :math:`a_0`
+    is the effective width of the linear region of the transformation.
+    In that region, the transformation is
+    :math:`a \\rightarrow a + {\\cal O}(a^3)`.
+    For large values of :math:`a` the transformation behaves as
+    :math:`a \\rightarrow a_0 \\, {\\rm sgn}(a) \\ln |a| + {\\cal O}(1)`.
+    """
+
+    name = 'asinh'
+
+    auto_tick_multipliers = {
+        3: (2, ),
+        4: (2, ),
+        5: (2, ),
+        8: (2, 4),
+        10: (2, 5),
+        16: (2, 4, 8),
+        64: (4, 16),
+        1024: (256, 512)
+    }
+
+    def __init__(self, axis, *, linear_width=1.0,
+                 base=10, subs='auto', **kwargs):
+        """
+        Parameters
+        ----------
+        linear_width : float, default: 1
+            The scale parameter (elsewhere referred to as :math:`a_0`)
+            defining the extent of the quasi-linear region,
+            and the coordinate values beyond which the transformation
+            becomes asympotically logarithmic.
+        base : int, default: 10
+            The number base used for rounding tick locations
+            on a logarithmic scale. If this is less than one,
+            then rounding is to the nearest integer multiple
+            of powers of ten.
+        subs : sequence of int
+            Multiples of the number base used for minor ticks.
+            If set to 'auto', this will use built-in defaults,
+            e.g. (2, 5) for base=10.
+        """
+        super().__init__(axis)
+        self._transform = AsinhTransform(linear_width)
+        self._base = int(base)
+        if subs == 'auto':
+            self._subs = self.auto_tick_multipliers.get(self._base)
+        else:
+            self._subs = subs
+
+    linear_width = property(lambda self: self._transform.linear_width)
+
+    def get_transform(self):
+        return self._transform
+
+    def set_default_locators_and_formatters(self, axis):
+        axis.set(major_locator=AsinhLocator(self.linear_width,
+                                            base=self._base),
+                 minor_locator=AsinhLocator(self.linear_width,
+                                            base=self._base,
+                                            subs=self._subs),
+                 minor_formatter=NullFormatter())
+        if self._base > 1:
+            axis.set_major_formatter(LogFormatterSciNotation(self._base))
+        else:
+            axis.set_major_formatter('{x:.3g}'),
```

Comment:
For my suggestion to leverage the SymmetricalLogLocator, I think something like the below suggestion looks pretty good in general (the `*2` seems to get rid of some of the overlapping near-zero tick texts). This would remove the need for the AsinhLocator and tests altogether by leveraging the SymLog work that finds the decades, but still keeps the subs info you put in to the scale. I do understand the hesitancy of using the older "threshold" and trying to get away from that, but I still think that the linear_width and threshold are very closely related ideas, so it isn't all that crazy to relate the two either.

```suggestion
        axis.set(major_locator=SymmetricalLogLocator(
                     base=self._base,
                     linthresh=2 * self.linear_width),
                 major_formatter=LogFormatterSciNotation(self._base),
                 minor_locator=SymmetricalLogLocator(
                     base=self._base,
                     linthresh=2 * self.linear_width,
                     subs=self._subs),
                 minor_formatter=NullFormatter())
```

Reply:
Thanks, @greglucas , for going through this again and for another set of helpful refinements.

The proposed removal of the `AsinhLocator` is obviously a more major decision, so I'd be interested in others' views before committing the last of your proposed changes. I'm obviously not impartial in my loyalty to `AsinhLocator`, or my reservations about the discontinuities that are an intrinsic part of the `SymmetricalLogLocator` that led to the creation of `AsinhScale`. There are imperfections and compromises in both of these approaches. Overall, I prefer the mathematical cleanliness of `asinh` for both the `Scale` and `Locator`.